js/pages/configuration/configuration.js

```diff
@@ -127,29 +164,52 @@ define([
       this.updateSourceDaimonPriority(selectedSource.sourceKey, 'Results');
       return true;
     };
-    
+
     checkSourceConnection(source) {
       sourceApi.checkSourceConnection(source.sourceKey)
         .then( ({ data }) =>
-           source.connectionCheck(data.sourceId === undefined ?
-               sourceApi.connectionCheckState.failed : sourceApi.connectionCheckState.success))
-        .catch(() => {source.connectionCheck(sourceApi.connectionCheckState.failed);});
-        source.connectionCheck(sourceApi.connectionCheckState.checking);
+          source.connectionCheck(data.sourceId === undefined ?
+            sourceApi.buttonCheckState.failed : sourceApi.buttonCheckState.success))
+        .catch(() => {source.connectionCheck(sourceApi.buttonCheckState.failed);});
+      source.connectionCheck(sourceApi.buttonCheckState.checking);
     };
-    
+
+    async refreshSourceCache(source) {
+      try {
+        const { data } = await sourceApi.refreshSourceCache(source.sourceKey);
```

Comment:
`checking` state should be set before the query is sent - otherwise, the button will remain blue until we receive a response which may take time and a user can be confused

Reply:
fixed